misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlLexer.java

```diff
@@ -200,6 +206,16 @@ private STToken readToken() {
         return token;
     }
 
+    private STToken readNewlineToken() {
+        reader.mark();
+        if (reader.isEOF()) {
+            return getSyntaxToken(SyntaxKind.EOF_TOKEN);
+        }
```

Comment:
Here its good to check whether the next character is a newline
- If so, send the newline token.
- Otherwise, go back to the default mode and send a normal token.

Reply:
I've done the same thing before submitting PR. It did not make any difference when I tested with different inputs as we are only starting the mode when processing newline. but yeah having the check is good for maintainability. Added this in the last commit.